src/utils.py

```diff
@@ -1,7 +1,37 @@
+import pandas as pd
+from random import sample
+from sklearn.preprocessing import LabelEncoder
+from torchtext.data.utils import get_tokenizer
+from torchtext.vocab import build_vocab_from_iterator
 
-# Data preprocessing stuff and loading in the data will go here.
-# 1. all lower case
-# 2. remove punctuation besides periods. Convert periods to a token like <PERIOD>
-# 3. Start of text and end of text token <S> <E>
-# 4. convert numbers to a number token <NUM>
-# 5. More stuff as we think of is
+def get_data_torchtext(file_path="./data/data.csv", sample_fraction=0.4, vocab_specials=["<unk>"]):
+    # Read CSV data
+    columns = ['target', 'ids', 'date', 'flag', 'user', 'text']
+    df = pd.read_csv(file_path, header=None, names=columns, encoding='ISO-8859-1')
+
+    # Extract relevant columns
+    df = df[["target", "text"]]
+
+    # Label encoding for target
+    label_encoder = LabelEncoder() 
+    df['target'] = label_encoder.fit_transform(df['target'])
+
+    # Convert to a list of tuples
+    data = list(zip(df["target"].tolist(), df["text"].tolist()))
+
+    # Shuffle and sample the data
+    data = sample(data, int(len(data) * sample_fraction))
```

Comment:
This line here means we are only taking some fraction of the dataset. We want to use the whole dataset ideally.

Reply:
Now, the function will use the entire dataset without sampling. I modified the get_data_torchtext function to remove the sampling step.